File: Backend/src/main/java/org/example/backend/domain/SubjectRequirement.java
package org.example.backend.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

@Getter
@Setter
@Entity
@Table(name = "subject_requirement")
public class SubjectRequirement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @ColumnDefault("nextval('subject_requirement_id_seq')")
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "field", nullable = false, length = 300)
    private String field;

    @Column(name = "grade", nullable = false)
    private Short grade;

    @Column(name = "period", nullable = false)
    private Short period;

    @Column(name = "advanced", nullable = false)
    private Boolean advanced = false;

}

File: TTSolver/Model/SubjectRequirement.py
class SubjectRequirement:
    def __init__(self, id: int, field: str, grade: int, period: int, advanced: bool):
        self.id = id
        self.field = field
        self.grade = grade
        self.period = period
        self.advanced = advanced

    def __str__(self):
        return f"SubjectRequirement(id={self.id}, field={self.field}, grade={self.grade}, period={self.period}, advanced={self.advanced})"
